roles/ceph-rgw-loadbalancer/tasks/pre_requisite.yml

```diff
@@ -1,11 +1,71 @@
 ---
+- name: assert that keepalived_version is defined
+  assert:
+    that:
+      - keepalived_version == 'distro' or keepalived_version is version('0.0.0', '>', strict=True)
+    fail_msg: keepalived_version must either be 'distro' or valid version number
+
 - name: install haproxy and keepalived
   package:
-    name: ['haproxy', 'keepalived']
+    name: haproxy
     state: present
   register: result
   until: result is succeeded
 
+- name: install keepalived
+  package:
+    name: keepalived
+    state: present
+  register: result
+  when: keepalived_version == 'distro'
+  until: result is succeeded
+
+- name: install non-distro version of keepalived
+  block:
+
+    - name: check if specified keepalived version already exists
+      command: "ls /usr/sbin/keepalived-{{ keepalived_version }}"
+      failed_when: false
+      changed_when: false
+      register: check_keepalived_version
```

Comment:
Nit-picking, but I'd move this after the **install required packages for keepalived** task, so it's immediately before tasks which run conditionally depending on its result.

Reply:
moved